hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/http/TestHttpServer.java

```diff
@@ -663,8 +662,7 @@ private HttpServer2 checkBindAddress(String host, int port, boolean findPort)
     HttpServer2 server = createServer(host, port);
     try {
       // not bound, ephemeral should return requested port (0 for ephemeral)
-      List<?> listeners = (List<?>) Whitebox.getInternalState(server,
-          "listeners");
+      List<?> listeners = server.getListeners();
       ServerConnector listener = (ServerConnector)listeners.get(0);
```

Comment:
We can use `List<ServerConnector>` instead of `List<?>` and then we can remove the casting `(ServerConnector)`.

Reply:
Thank you very much for your help reviewing the code, I will modify the code.